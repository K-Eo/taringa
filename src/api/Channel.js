import APIBase from './APIBase'

class Channel extends APIBase {
  about(id) {
    return super.execute(`/c/${id}/about`)
  }
}

export default Channel
