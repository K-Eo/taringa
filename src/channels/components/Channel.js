import React, { Component } from 'react'

import withFilter from '../../HOC/Filterable'
import FeedRoute from '../../feed/components/FeedRoute'
import ChannelCard from './ChannelCard'

class Channel extends Component {
  render() {
    const { channel } = this.props.match.params

    return (
      <div className="row">
        <div className="col-8">
          <FeedRoute
            exact
            id={`${channel}.hot`}
            path={`/c/${channel}`}
            url={`/c/${channel}/feed`}
          />
          <FeedRoute
            id={`${channel}.recents`}
            path={`/c/${channel}/recents`}
            url={`/c/${channel}/feed`}
          />
          <FeedRoute
            id={`${channel}.tops`}
            path={`/c/${channel}/tops`}
            url={`/c/${channel}/tops/week`}
          />
        </div>
        <div className="col-4">
          <ChannelCard />
        </div>
      </div>
    )
  }
}

const mapFilters = props => {
  const channel = props.match.params.channel

  return {
    hot: {
      displayName: 'Destacados',
      exact: true,
      id: 'hot',
      pathname: `/c/${channel}`,
    },
    recents: {
      displayName: 'Recientes',
      exact: false,
      id: 'recents',
      pathname: `/c/${channel}/recents`,
    },
    tops: {
      displayName: 'Tops',
      exact: false,
      id: 'tops',
      pathname: `/c/${channel}/tops`,
    },
  }
}

export default withFilter(mapFilters)(Channel)
