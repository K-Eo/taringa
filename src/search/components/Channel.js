import React from 'react'
import { Link } from 'react-router-dom'

import CardDecorator from '../../common/CardDecorator'

const Channel = props => {
  return (
    <div className="col-6 mb-4">
      <CardDecorator
        avatar={props.thumbnail}
        cover={props.background}
        placeholder={false}
      >
        <div className="card-body">
          <Link to={`/c/${props.name}`} onClick={() => window.scrollTo(0, 0)}>
            <h5 className="card-title">{props.title}</h5>
          </Link>

          {props.description !== props.title && (
            <p className="card-text mt-3 mb-4">{props.description}</p>
          )}
        </div>
      </CardDecorator>
    </div>
  )
}

export default Channel
