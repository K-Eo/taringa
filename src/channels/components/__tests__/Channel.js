import React from 'react'

import Channel from '../Channel'

describe('Channel', () => {
  it('exists', () => {
    expect(Channel).toBeDefined()
  })

  it('renders', () => {
    const match = {
      params: {
        channel: 'foo',
        filter: 'hot',
      },
    }

    const wrapper = shallow(<Channel match={match} />)

    expect(wrapper).toMatchSnapshot()
  })
})
