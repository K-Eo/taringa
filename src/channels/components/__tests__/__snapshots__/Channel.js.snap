// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Channel renders 1`] = `
<div
  className="row"
>
  <Connect(Filterable)
    filters={
      Object {
        "hot": Object {
          "displayName": "Destacados",
          "exact": true,
          "id": "hot",
          "pathname": "/c/foo",
          "url": "/c/foo/feed",
        },
        "recents": Object {
          "displayName": "Recientes",
          "exact": false,
          "id": "recents",
          "pathname": "/c/foo/recents",
          "url": "/c/foo/feed",
        },
        "tops": Object {
          "displayName": "Tops",
          "exact": false,
          "id": "tops",
          "pathname": "/c/foo/tops",
          "url": "/c/foo/tops/week",
        },
      }
    }
  />
  <div
    className="col-8"
  >
    <Connect(Summaries)
      feedId="foo"
      filter="hot"
      url="/c/foo/feed"
    />
  </div>
  <div
    className="col-4"
  >
    <withRouter(Connect(ResourceHOC)) />
  </div>
</div>
`;
